lib/src/signed_video_h26x_common.c

```diff
@@ -884,12 +890,17 @@ update_linked_hash(signed_video_t *self, uint8_t *hash, size_t hash_size)
     if (hash_size != self->sign_data->hash_size) return SV_INVALID_PARAMETER;
   }
   gop_info_t *gop_info = self->gop_info;
-  uint8_t *stored_hash = &gop_info->linked_hashes[hash_size];
-  uint8_t *linked_hash = &gop_info->linked_hashes[0];
-  // Move the content of stored_hash to linked_hash
-  memmove(linked_hash, stored_hash, hash_size);
-  // Copy the new hash to stored_hash
-  memcpy(stored_hash, hash, hash_size);
+  uint8_t *new_hash = &gop_info->linked_hashes[hash_size * 2];  // Newest linked hash
+  uint8_t *middle_hash = &gop_info->linked_hashes[hash_size];  // Middle hash in the chain
+  uint8_t *oldest_hash = &gop_info->linked_hashes[0];  // Oldest hash in the chain
```

Comment:
Note that this buffer does not have all zeros from the beginning, which will make verifying the linked hash fail the first 2 times of a stream. When building in debug mode, which is what we do in development they are usually set to zero, but not when building in release mode. Valgrind will detect that and you can see it in the meson build test of this PR.